src/python/WMCore/WMException.py

```diff
@@ -114,6 +116,13 @@ def addInfo(self, **data):
         exception instance
         """
         for key, value in viewitems(data):
+            # unicode sandwich: convert sequence of bytes strings to
+            # unicode codepoints strings as soon as possible
+            # assumption: value is not iterable (list, dict, tuple, ...)
```

Comment:
@amaltaro is this assumption reasonable?

What I mean is that if `data` is like 

```python
data = {
key: 3.14,
key1: "string"
}
```
then we are ok, but if it can be
```python
data = {
top_key: 3.14,
top_key1: {
  nested_key : "string"
}
}
```

then we can incur in the same bug. The solution would be changing `self.addInfo()` so that it is recursive and can traverse a generic nested dictionary looking for bytes strings and converting them to unicode strings. This would add some complexity and it is why I preferred not to go down this path without checking with you before

Reply:
I suspect `data.values()` will always be a string, but I'm not 100% sure. 
I guess there is no need to make it recursive, if we happen to have another issue like this, then we know where implementation is required.